### PR TITLE
Use the new django-oscar-extensions overrides and data migrations.

### PR DESCRIPTION
**Pull request by stephensanchez**
@dianakhuang @rlucioni @wedaly 

This allows us to wire in the django-oscar-extensions project along with the new data migrations that set up our product catalog. 

With this change, we can install pip requirements, sync up database tables, and have our catalog pre-installed with the demo course.

### CODE REVIEW

**Comment by wedaly** [comment]
Should `ecommerce/fixtures/catalog_definition.json` be deleted now?  Otherwise, lgtm.

**Comment by rlucioni** [comment]
:+1:

**Comment by stephensanchez** [comment]
Oh, I thought I deleted the fixture. Hrmph. Yeah I'll make sure that's gone.

File: ecommerce/settings/base.py [hunk 1]
```diff
@@ -240,7 +240,13 @@
 ]
 
 # See: https://docs.djangoproject.com/en/dev/ref/settings/#installed-apps
-INSTALLED_APPS = DJANGO_APPS + LOCAL_APPS + get_core_apps()
+INSTALLED_APPS = DJANGO_APPS + LOCAL_APPS + get_core_apps(
+    [
+        'extensions.apps.catalogue',
+        'extensions.apps.partner',
+        'extensions.apps.orders'
```
Comment (rlucioni) [inline ecommerce/settings/base.py]: These will need to be changed to `'extensions.<appname>'` to reflect the restructuring of the django-oscar-extensions project.

Comment (stephensanchez) [inline ecommerce/settings/base.py]: yep!

**Pull request merged**